fw/preamp/src/ports.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * AmpliPi Home Audio
+ * Copyright (C) 2021 MicroNova LLC
+ *
+ * Port usage and functions for GPIO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "ports.h"
+#include "stm32f0xx.h"
+
+static GPIO_TypeDef * getPort(Pin pp){
+	switch(pp.port){
+	case 'A':
+		return GPIOA;
+	case 'B':
+		return GPIOB;
+	case 'C':
+		return GPIOC;
+	case 'D':
+		return GPIOD;
+	case 'F':
+		return GPIOF;
+	default:
+		while(1);
```

Comment:
This should be removed in production. I would rather return an invalid port then halt.